### PR TITLE
rename output files to match pseudopeople dataset names

### PR DESCRIPTION
**Pull request by rmudambi**
### Description

- *Category*: bugfix
- *JIRA issue*: [MIC-4020](https://jira.ihme.washington.edu/browse/MIC-4020)
- *Research reference*: 

### Changes and notes
Rename the output files to match pseudopeople dataset names
Introduce DatasetNames constants and replace all naked string references to them.

### Verification and Testing
CI passes
Ran psimulate and make_results

### CODE REVIEW

File: src/vivarium_census_prl_synth_pop/components/observers.py [hunk 1]
```diff
@@ -84,6 +84,11 @@ def input_values(self):
     def output_columns(self):
         pass
 
+    @property
+    @abstractmethod
+    def output_name(self) -> str:
```
Comment (stevebachmeier) [inline src/vivarium_census_prl_synth_pop/components/observers.py:89]: Funny b/c I'm pretty sure this was the original implementation we proposed. Can't recall why this got axed

File: src/vivarium_census_prl_synth_pop/constants/metadata.py [hunk 2]
```diff
@@ -275,3 +275,16 @@ class __Scenarios(NamedTuple):
 }
 
 SUPPORTED_EXTENSIONS = ["hdf", "parquet"]
+
+
+class DatasetNames:
+    """Container for Form names"""
```
Comment (stevebachmeier) [inline src/vivarium_census_prl_synth_pop/constants/metadata.py:281]: "Dataset" names

You could also import from pseudopeople instead?

Comment (stevebachmeier) [reply src/vivarium_census_prl_synth_pop/constants/metadata.py:281]: Oh, nm, it's kinda different w/ taxes.

**Review by stevebachmeier — APPROVED** [review]

File: src/vivarium_census_prl_synth_pop/constants/metadata.py (hunk 2 above)
Comment (rmudambi) [reply src/vivarium_census_prl_synth_pop/constants/metadata.py:281]: I decided that was out of scope for this.

**Pull request merged**